plugin-maven/src/main/java/com/diffplug/spotless/maven/MavenProvisioner.java

```diff
@@ -27,9 +25,7 @@ private MavenProvisioner() {}
 
 	public static Provisioner create(ArtifactResolver artifactResolver) {
 		Objects.requireNonNull(artifactResolver);
-
-		return mavenCoords -> mavenCoords.stream()
-				.flatMap(coord -> artifactResolver.resolve(coord).stream())
-				.collect(toSet());
+		return mavenCoords -> artifactResolver.resolve(mavenCoords);
```

Comment:
@lutovich Sorry, maybe it's just to late... I lost you. Can you explain in detail?

Reply:
I just mean that lambda

```java
return mavenCoords -> artifactResolver.resolve(mavenCoords);
```

can probably be replaced with method reference

```java
return artifactResolver::resolve;
```

Then `Objects.requireNonNull(artifactResolver);` can also be removed because method ref will fail with NPE anyway.